src/fsharp/TypeChecker.fs

```diff
@@ -2112,9 +2112,9 @@ module GeneralizationHelpers =
             match tp.Constraints |> List.partition (function (TyparConstraint.CoercesTo _) -> true | _ -> false) with 
             | [TyparConstraint.CoercesTo(cxty,_)], others -> 
                  // Throw away null constraints if they are implied 
-                 match others |> List.filter (function (TyparConstraint.SupportsNull(_)) -> not (TypeSatisfiesNullConstraint cenv.g m cxty) | _ -> true) with
-                 | [] -> Some cxty
-                 | _ -> None
+                 if others |> List.exists (function (TyparConstraint.SupportsNull(_)) -> not (TypeSatisfiesNullConstraint cenv.g m cxty) | _ -> true) 
```

Comment:
would we prefer this style:

```fsharp
// Throw away null constraints if they are implied 
let hasTypesWithNullConstraint =
    others 
    |> List.exists (
        function
        | (TyparConstraint.SupportsNull(_)) -> not (TypeSatisfiesNullConstraint cenv.g m cxty) 
        | _ -> true
    ) 
if hasTypesWithNullConstraint 
then None
else Some cxty
```

I know most of the compiler has long lines, but from my perspective it really ease the flow of reading when we adopt shorter lines / expressions, introducing / naming a variable also helps